File: Games/Drip Drop/Assets/Scripts/Play/Destroy.cs
using UnityEngine;
using System.Collections;

public class Destroy : MonoBehaviour {
	private int lives = 0;
	public TextMesh Buckets;
	public TextMesh Score;
	public GoogleAnalyticsV4 googleAnalytics;
	public Destroy_Score Destroy1;

	void OnTriggerEnter2D(Collider2D collisionObject) {

		if (collisionObject.gameObject.tag == "Driplet") {
						Destroy (collisionObject.gameObject);
			     if (PlayerPrefs.GetInt ("Lives-Left") > 0) {
				 Destroy (collisionObject.gameObject);
				 lives = PlayerPrefs.GetInt ("Lives-Left");
				 lives = lives - 1;
				 PlayerPrefs.SetInt ("Lives-Left", lives);
				 PlayerPrefs.Save ();
				 Buckets.text = ("Lives: " + PlayerPrefs.GetInt("Lives-Left"));
			     }
			     else if (PlayerPrefs.GetInt ("Lives-Left") == 0) {
						        if (PlayerPrefs.GetInt ("GameScore") > PlayerPrefs.GetInt ("HighScore")) {
								PlayerPrefs.SetInt ("HighScore", PlayerPrefs.GetInt ("GameScore"));
								PlayerPrefs.Save ();
						        }
						        Time.timeScale = 1.0f;
				                Destroy (collisionObject.gameObject);
								googleAnalytics.LogScreen("Died: " + Score.text.Substring(7));
								Destroy1.bannerView.Destroy();
						        Application.LoadLevel (2);
			                    }
				}
		 else if (collisionObject.gameObject.tag == "Dirty") {
						Destroy (collisionObject.gameObject);
				}
		 else if (collisionObject.gameObject.tag == "Live") {
			Destroy (collisionObject.gameObject);
		}
	}
}

File: Games/Jump the Box/Assets/Scripts/Game/Destroy.js
#pragma strict
var Player1 : GameObject;
var Player2 : GameObject;
var Highscore : int = 0;

function OnTriggerEnter2D(other : Collider2D) {
if (other.gameObject.tag == "Object") {
Destroy(other.gameObject);
Player1.GetComponent(BoxCollider2D).enabled = true;
Player2.GetComponent(BoxCollider2D).enabled = true;
Player1.transform.position = new Vector3 (-1.5, -4, -7);
Player2.transform.position = new Vector3 (1.5, -4, -7);
}
if (other.gameObject.tag == "Player1") {
Time.timeScale = 1;
var Highscore0 = PlayerPrefs.GetInt("Highscore0");
PlayerPrefs.SetInt("Highscore1", Highscore0);
PlayerPrefs.SetInt("PlayGame", 10);
PlayerPrefs.Save();
if(PlayerPrefs.GetInt("Highscore0") >= 10){
PlayerPrefs.SetInt("Get10Points", 10);
PlayerPrefs.Save();
}
if(PlayerPrefs.GetInt("Highscore0") >= 20){
PlayerPrefs.SetInt("Get20Points", 10);
PlayerPrefs.Save();
}
if(PlayerPrefs.GetInt("Highscore0") >= 50){
PlayerPrefs.SetInt("Get50Points", 10);
PlayerPrefs.Save();
}
if(PlayerPrefs.GetInt("Highscore0") >= 100){
PlayerPrefs.SetInt("Get100Points", 10);
PlayerPrefs.Save();
}
Highscore = PlayerPrefs.GetInt("Highscore");
if (Highscore0 > Highscore){
PlayerPrefs.SetInt("Highscore", Highscore0);
PlayerPrefs.Save();
}
Application.LoadLevel(0);
}
if (other.gameObject.tag == "Player2") {
Time.timeScale = 1;
var Highscore06 = PlayerPrefs.GetInt("Highscore0");
PlayerPrefs.SetInt("Highscore1", Highscore06);
PlayerPrefs.SetInt("PlayGame", 10);
PlayerPrefs.Save();
if(PlayerPrefs.GetInt("Highscore0") >= 10){
PlayerPrefs.SetInt("Get10Points", 10);
PlayerPrefs.Save();
}
if(PlayerPrefs.GetInt("Highscore0") >= 20){
PlayerPrefs.SetInt("Get20Points", 10);
PlayerPrefs.Save();
}
if(PlayerPrefs.GetInt("Highscore0") >= 50){
PlayerPrefs.SetInt("Get50Points", 10);
PlayerPrefs.Save();
}
if(PlayerPrefs.GetInt("Highscore0") >= 100){
PlayerPrefs.SetInt("Get100Points", 10);
PlayerPrefs.Save();
}
Highscore = PlayerPrefs.GetInt("Highscore");
if (Highscore06 > Highscore){
PlayerPrefs.SetInt("Highscore", Highscore0);
PlayerPrefs.Save();
}
Application.LoadLevel(0);
}
}